lib/bindings/HTTPBinding.js

```diff
@@ -127,7 +127,9 @@ function parseData(req, res, next) {
         next(error);
     } else {
         req.jsonPayload = data;
-
+        if (req.body !== undefined || req.body === Object || Buffer.isBuffer(req.body)) {
```

Comment:
Could you elaborate why is required this condition? (specially why using OR operator)

Reply:
Hi @mapedraza , we require the condition because test case was failing in commands-Polling-test.js for undefind data.

Reply:
Tests are failling even after restoring this line. Maybe you should restore this line too:
https://github.com/telefonicaid/iotagent-json/blob/bf45b6b74fce46bd3d43eecb6ea6e31208ebcf99/lib/bindings/HTTPBinding.js#L293

Additionaly, could you explain why is failing and why is required? I do not understand why the 3 conditions are nested with OR operator.

Thanks

Reply:
Hi @mapedraza, I have run the test files on my local system and all the test case were running properly and passing. Here you can find in the screenshot. 
Please guide me how can I fix it here?
![Screenshot 2023-08-16 124608](https://github.com/telefonicaid/iotagent-json/assets/102344018/0988fe55-0739-4207-8c78-9284dfb4fe09)

Reply:
> Additionaly, could you explain why is failing and why is required? I do not understand why the 3 conditions are nested with OR operator.
I have resolved it by latest changes , as per the requirements